src/code/audio_heap.c

```diff
@@ -1009,40 +1009,42 @@ void AudioHeap_Init(void) {
         reverb->tunedSample.tuning = 1.0f;
         reverb->sample.codec = CODEC_REVERB;
         reverb->sample.medium = MEDIUM_RAM;
-        reverb->sample.size = reverb->windowSize * 2;
+        reverb->sample.size = reverb->windowSize * SAMPLE_SIZE;
         reverb->sample.sampleAddr = (u8*)reverb->leftRingBuf;
         reverb->loop.start = 0;
         reverb->loop.count = 1;
         reverb->loop.end = reverb->windowSize;
 
         if (reverb->downsampleRate != 1) {
             reverb->unk_0E = 0x8000 / reverb->downsampleRate;
-            reverb->unk_30 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, 0x20);
-            reverb->unk_34 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, 0x20);
-            reverb->unk_38 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, 0x20);
-            reverb->unk_3C = AudioHeap_AllocZeroed(&gAudioContext.miscPool, 0x20);
+            reverb->unk_30 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, SAMPLES_PER_FRAME * SAMPLE_SIZE);
```

Comment:
The size here matches up with `sizeof(ADPCMSTATE)` from abi.h, not sure if it's applicable however?
Are the types in abi.h usable anywhere? I think args to the abi commands should use these where possible since the buffer sizes expected by these are baked into the microcode, although this is probably out of scope.

Reply:
I see, `SAMPLES_PER_FRAME * SAMPLE_SIZE` is quite common which has the same value as `sizeof(ADPCMSTATE)`. Would you think it would be better to extend `sizeof(ADPCMSTATE)` to most cases of `sizeof(ADPCMSTATE)`?

But also, there are some sizes that would be nice to have that I don't see in `abi.h`, for example, filters come in groups of 8 samples but right now, the 8 remains hardcoded as there isn't anything in `abi.h` for them